### PR TITLE
[fix #158] Add changefeed start/end

### LINKED ISSUE #158: cdc: Support specifying start-key & end-key of changefeed
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

Support specifying `start-key` & `end-key` of changefeed, to replication some range of keys other than all.

Arguments
* format: 
  * "raw": The input string is directly encoded as a key in binary format.
  * "hex": The default encoding format. The input string is treated as a hexadecimal number.
  * "escaped": First escape (backslash) the input string, and then encode it into binary format, for example, abc\xFF\x00\r\n.
* start-key: inclusive start of range.
* end-key: exclusive end of range.

(The arguments design refers to https://docs.pingcap.com/tidb/stable/rawkv-backup-and-restore.)

Implementation hint: [ParseKey](https://github.com/tikv/migration/blob/7d32868592f9a7ea6392cb5776ea3f4c28d4c767/br/pkg/utils/key.go#L25).

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by haojinming**
Signed-off-by: haojinming <jinming.hao@pingcap.com>

changefeed start/end

### What problem does this PR solve?

Issue Number: close #158 

Problem Description: 

### What is changed and how does it work?
 support specify `start_key/end_key/format` when create changefeed with `tikv-cdc cli changefeed create`

### Code changes

- Has exported function/method change
- Has exported variable/fields change
- Has persistent data change

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test
- Manual test (add detailed scripts or steps below)

### Side effects

- No related changes
-
### Related changes

- No related changes

### CODE REVIEW

**Review by pingyu — COMMENTED** [review]
I think we can allow to update start & end key of changefeed.

File: cdc/cdc/model/changefeed.go [hunk 1]
```diff
@@ -114,6 +114,12 @@ type ChangeFeedInfo struct {
 	StartTs uint64 `json:"start-ts"`
 	// The ChangeFeed will exits until sync to timestamp TargetTs
 	TargetTs uint64 `json:"target-ts"`
+	// The Start Key of changefeed, inclusive
+	StartKey string `json:"start-key"`
+	// The End Key of changefeed, exclusive
+	EndKey string `json:"end-key"`
+	// Format of StartKey and EndKey, "raw", "escaped", "hex"
```
[the remark this answers is not in the document]
Comment (haojinming) [reply cdc/cdc/model/changefeed.go:121]: The problem is that when user query changefeed, do we need show the start/end keys exactly same with user's input?

[the remark this answers is not in the document]
Comment (pingyu) [reply cdc/cdc/model/changefeed.go:121]: I think... yes.
OK I see why we keep the original keys & format. Then I suggest to leave some comment about this.
Besides, suggest to check format & keys in `createChangefeedOptions.validate()`.

**Review by zeminzhou — APPROVED** [review]

File: cdc/cdc/owner/scheduler_v1_test.go [hunk 2]
```diff
@@ -121,7 +121,7 @@ func (s *schedulerSuite) TestScheduleOneCapture(c *check.C) {
 	s.addCapture(captureID)
 
 	// add 4 keyspans
-	s.scheduler.updateCurrentKeySpans = func(ctx cdcContext.Context) ([]model.KeySpanID, map[model.KeySpanID]regionspan.Span, error) {
+	s.scheduler.updateCurrentKeySpans = func(ctx cdcContext.Context, info *model.ChangeFeedInfo) ([]model.KeySpanID, map[model.KeySpanID]regionspan.Span, error) {
```
[the remark this answers is not in the document]
Comment (haojinming) [reply cdc/cdc/owner/scheduler_v1_test.go:125]: unit test is added.

File: cdc/pkg/util/key_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		},
+		{
+			Format: "hex",
+			Key:    "78bcacef",
+			Expect: []byte{0x78, 0xbc, 0xac, 0xef},
+			Valid:  true,
+		},
+		{
+			Format: "escaped",
+			Key:    "abcde",
```
[the remark this answers is not in the document]
Comment (haojinming) [reply cdc/pkg/util/key_test.go]: done

File: cdc/pkg/cmd/cli/cli_changefeed_create.go [hunk 4]
```diff
@@ -72,6 +82,10 @@ func (o *changefeedCommonOptions) addFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&o.sortEngine, "sort-engine", model.SortUnified, "sort engine used for data sort")
 	cmd.PersistentFlags().StringVar(&o.sortDir, "sort-dir", "", "directory used for data sort")
 	_ = cmd.PersistentFlags().MarkHidden("sort-dir")
+	cmd.PersistentFlags().StringVar(&o.format, flagKeyFormat, "hex", "The format of start and end key. Available options: \"raw\", \"escaped\", \"hex\".")
+	cmd.PersistentFlags().StringVar(&o.startKey, flagStartKey, "", "The start key of the changefeed, key is inclusive.")
+	cmd.PersistentFlags().StringVar(&o.endKey, flagEndKey, "", "The end key of the changefeed, key is exclusive.")
+
```
Comment (zeminzhou) [inline cdc/pkg/cmd/cli/cli_changefeed_create.go:88]: If user don't input 'start-key' and 'end-key', can we get a defaut 'start-key' and 'end-key'?

Comment (haojinming) [reply cdc/pkg/cmd/cli/cli_changefeed_create.go:88]: Default is empty and we encode it to [r000, r001) here: https://github.com/haojinming/migration/blob/6f40097eedf69f64e50179d58776ff3810f28976/cdc/pkg/util/key.go#L27

**Review by pingyu — APPROVED** [review]
Great ! LGTM~

**Pull request merged**